Add step-by-step explanation for plugin install flow (#3599)
openbao/openbao: website/content/docs/configuration/plugins.mdx

# Declarative plugins

[External plugins](../plugins/index.mdx) can be declared with the `plugin` stanza.
This is primarily intended for:

1. **Simple plugin deployment:**
   The combination of OCI-based plugin distribution, auto-download, and auto-registration
   is recommended plugin deployment flow.
2. **`kms` plugin deployment:**

>    `kms` plugin cannot be registered via the API (since they must be available in a sealed state).

Reviewer: ```suggestion
   `kms` plugins cannot be registered via the API (since they must be available in a sealed state).
```

>    The declarative method is currently the only way to define `kms` plugins.

Reviewer 2: ```suggestion
   The declarative method is the only way to define `kms` plugins.
```

Reviewer 2: I don't imagine this changes in the future.

> The `plugin` stanza allows the declaration of both OCI images and manually downloaded plugins.

Reviewer: ```suggestion
The `plugin` stanza allows the declaration of both plugins automatically downloaded from an OCI registry and manually downloaded plugins.
```

# Declarative plugins

[External plugins](../plugins/index.mdx) can be declared with the `plugin` stanza.
This is primarily intended for:

1. **Simple plugin deployment:**
   The combination of OCI-based plugin distribution, auto-download, and auto-registration

>    is recommended plugin deployment flow.

Reviewer: ```suggestion
   is the recommended plugin deployment flow.
```

2. **`kms` plugin deployment:**
   `kms` plugins cannot be registered via the API (since they must be available in a sealed state).
   The declarative method is the only way to define `kms` plugins.

The `plugin` stanza allows the declaration of both plugins automatically
downloaded from an OCI registry and manually downloaded plugins.

## OCI-based plugin distribution

OpenBao supports the distribution of plugins in Open Container Initiative (OCI)

> images. Operators can define the desired plugins for their OpenBao cluster

Reviewer 3: _Optional extra cleanup_
```suggestion
artefacts. Operators can define the desired plugins for their OpenBao cluster
```

Even if the artefacts are literally also images, say artefacts.

Author: I moved this to a separate MR: https://github.com/openbao/openbao/pull/3630

}
```

> Or it specifies a plugin to automatically register that was manually downloaded:

Reviewer 3: lowercase “or” looks ore idiomatic to me.

Author: To me, uppercase looks better 😄 
It feels like a new sentence, especially given the interruption by the code snippet. It makes it look like a new paragraph, and starting a paragraph with lowercase feels wrong.
